prover/src/shared_state.rs

```diff
@@ -116,12 +115,15 @@ impl SharedState {
         // instead.
 
         let _pending_task = pending_task.clone();
+        let _self = self.clone();
```

Comment:
Usually the underscore prefix is used for unused variables, but I see that this variable is being used.  Maybe rename it to `self_copy` or something similar?
Same with `_pending_task`.

Reply:
Thanks! This is fixed now